### PR TITLE
[5.1] Rewrite treeselect in vanilla JS

### PR DESCRIPTION
**Pull request by C-Lodder**
### Summary of Changes

This PR removes the jQuery dependency for the treeselect Javascript.

Also uses a `<div>` for the dropdown heading, as opposed to a `<h1>`.

### Testing Instructions

1. Go to edit any module in the Module Manager.
2. Navigate to the "Menu Assignment" tab
3. Select "Only on the pages selected"
4. Play around with all the treeselect options (expanding, collapsing, checking, unchecking, etc)

### CODE REVIEW

**Comment by fgsw** [comment]
I have tested this item :white_check_mark: successfully on 29ae5d3b5539d7d2e7b3c206366981edd7d5cd49<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42776">issues.joomla.org/tracker/joomla-cms/42776</a>.</sub>

File: build/media_source/system/js/treeselectmenu.es6.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
+/**
+ * @copyright  (C) 2012 Open Source Matters, Inc. <https://www.joomla.org>
+ * @license    GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+const treeselectmenu = document.getElementById('treeselectmenu').innerHTML;
+const direction = (document.dir !== undefined) ? document.dir : document.getElementsByTagName('html')[0].getAttribute('dir');
```
Comment (dgrammatiko) [inline build/media_source/system/js/treeselectmenu.es6.js]:
```suggestion
const direction = (document.dir !== undefined) ? document.dir : document.documentElement.dir;
```

**Review by dgrammatiko — CHANGES_REQUESTED** [review]

**Comment by Fedik** [comment]
I have tested this item :white_check_mark: successfully on 4f775be8a236528adfa311277ad213015c074f27<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42776">issues.joomla.org/tracker/joomla-cms/42776</a>.</sub>

**Comment by Fedik** [comment]
r2c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42776">issues.joomla.org/tracker/joomla-cms/42776</a>.</sub>

**Comment by LadySolveig** [comment]
Thank you for your work @C-Lodder I tested this and as soon as an individual menu item has been manually selected/deselected, the dropdown next to it for the entire menu for "selecting/deselecting" all menu items no longer has any effect on the previously manually changed entry. 
Only again after the complete page refresh.
However, this error already exists in the old version.
Could you please take a look here?

**Comment by C-Lodder** [comment]
@LadySolveig Can do, but the toolbar in the edit view is missing. I this a known bug with the recent upmerge?

**Comment by Fedik** [comment]
> but the toolbar in the edit view is missing

Clear the cache for autoload_psr4.php

**Comment by C-Lodder** [comment]
@LadySolveig Good catch. Fixed now

**Comment by LadySolveig** [comment]
Thank you @C-Lodder for your quick response and the fix. 
I'm very happy that you were able to fix this bug that has been around for a while!

**Pull request merged**